esphome/__main__.py

```diff
@@ -889,7 +889,7 @@ def parse_args(argv):
         "--port",
         help="The HTTP port to open connections on. Defaults to 6052.",
         type=int,
-        default=6052,
+        default=0,
```

Comment:
Untested, but i'm sure that this is really all that is needed for this to work
```suggestion
        default=os.getenv("ESPHOME_DASHBOARD_PORT", 6052),
```

Reply:
Updated according to your suggestion. Tested and it works: 
![image](https://github.com/user-attachments/assets/ba5c3249-8a85-43a4-81df-65bae60cc471)